packages/runtime/src/templates/server.ts

```diff
@@ -54,11 +54,14 @@ const getNetlifyNextServer = (NextServer: NextServerType) => {
           return
         }
 
-        // force Next to revalidate all ODB revalidation requests so that we always have fresh content...
-        // setting this header to the preview mode ID is how Next knows to revalidate
+        // force Next to revalidate all requests so that we always have fresh content
+        // for our ODBs and middleware is disabled at the origin
+        // but ignore in preview mode (prerender_bypass is set to true in preview mode)
+        // because otherwise revalidate will override preview mode
         // eslint-disable-next-line no-underscore-dangle
-        if (headers['x-nf-builder-cache'] === 'revalidate' && !headers.__prerender_bypass) {
-          console.log(`Revalidating ${url}`)
+        if (!headers.__prerender_bypass) {
```

Comment:
Gah, I realized last night that if we only revalidate when an ODB sends that header, then middleware will still be enabled for all other requests (because we're now relying on revalidate mode to disable middleware at the origin). This commit reverts that, so we now revalidate on every request and will have to find a different way to serve the initial static content (but perhaps that's for a separate PR, like you said).